File: ClassesDemo/src/Constraint.java
import java.time.LocalDateTime;

enum constraintType {
    Studies,
    Test,
    Work,
    Event,
    Rest,
    Other
}

public class Constraint extends Period {
    private constraintType type;

    public Constraint(String description, constraintType type, LocalDateTime startTime, LocalDateTime endTime) {
        super(description);
        this.type = type;
        this.startTime = startTime;
        this.endTime = endTime;

        updateHours();
    }

    // ===========================================================================

    public constraintType getType() {
        return type;
    }

    // ===========================================================================

    public void setType(constraintType type) {
        this.type = type;
    }

    // ===========================================================================

    @Override
    public String toString() {
        return ">> Constraint:\n"
                + "Description: " + description + "\n"
                + "Type: " + type + "\n"
                + "Start Time: " + startTime.format(FORMATTER) + "\n"
                + "End Time: " + endTime.format(FORMATTER) + "\n"
                + "Duration In Hours: " + String.format("%.2f", hours) + "\n"
                + "===================================\n";
    }

    // ===========================================================================
}

File: Server/src/services/constraint.ts
import { Period } from './period'

enum constraintType {
    Studies,
    Test,
    Work,
    Event,
    Rest,
    Other,
}

interface Constraint extends Period {
    type: constraintType
}

const printConstraint = (constraint: Constraint): void => {
    console.log('>> Constraint:')
    console.log('ID: ' + constraint.id)
    console.log('Description: ' + constraint.description)
    console.log('Type: ' + constraint.type)
    console.log('Start Time: ' + constraint.startTime)
    console.log('End Time: ' + constraint.endTime)
    console.log('Duration: ' + constraint.hours + ' Hours')
}

export { Constraint, constraintType, printConstraint }
